### ISSUE TITLE
Change mode dialog in automation/script editors - doesn't always save number fields

### ISSUE THREAD

**Issue by SeanPM5** [home-assistant/frontend#20978]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.
- [X] I have tried reproducing the issue in [safe mode](https://www.home-assistant.io/blog/2023/11/01/release-202311/#restarting-into-safe-mode) to rule out problems with unsupported custom resources.

### Describe the issue you are experiencing

In the "Change Mode" dialog, the numeric values in the `queue length` and `max number of parallel runs` fields do not get saved when using keyboard (enter/return key). Only when you manually press the "Change Mode" button with mouse.

<img width="441" alt="Screen Shot 2024-06-03 at 4 44 13 PM" src="https://github.com/home-assistant/frontend/assets/9065557/655cd809-2514-42e3-8d9b-e548560248d0">

Attached screen recording to demonstrate the issue and explained what's going on:
- Video starts off in single mode (default)
- Changed to Queued mode with queue length of `20` and saved by pressing "Change Mode" dialog button. It works fine when pressing this button.
- Changed queue length to `100` and saved by pressing enter key on keyboard. Dialog gets dismissed without error, so it looks like it works, but it doesn't actually save value and keeps previous value of `20`
- Changed mode to Parallel with 50 runs. Saved by pressing "Change Mode" dialog button with mouse. It works.
- Changed `max number of parallel runs` value to 20 and saved via enter key on keyboard. Doesn't work, still keeps previous value of `50`

So basically works via mouse but not keyboard.

https://github.com/home-assistant/frontend/assets/9065557/2c1995b4-1231-4468-9d11-4716b13371e8

### Describe the behavior you expected

Save values using keyboard to enter/return

### Steps to reproduce the issue

Listed above

### What version of Home Assistant Core has the issue?

2024.6.0b6
